pinot-spi/src/main/java/org/apache/pinot/spi/accounting/ExecutionContext.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.accounting;
+
+public interface ExecutionContext {
+
+   /**
+    * get query id of the execution context
+    * @return query id in string
+    */
+   String getQueryId();
+
+   /**
+    *
+    * @return get the root thread of execution context
+    */
+   Thread getRootThread();
```

Comment:
I am guessing this means runner thread ?

Also, when we want to extend this to broker where we currently don't have a concept of worker thread and runner thread, this rootThread API will continue to work I guess ?